app/lib/activitypub/activity/create.rb

```diff
@@ -237,8 +237,11 @@ def process_attachments
       next if attachment['url'].blank? || media_attachments.size >= 4
 
       begin
-        href             = Addressable::URI.parse(attachment['url']).normalize.to_s
-        media_attachment = MediaAttachment.create(account: @account, remote_url: href, description: attachment['summary'].presence || attachment['name'].presence, focus: attachment['focalPoint'], blurhash: supported_blurhash?(attachment['blurhash']) ? attachment['blurhash'] : nil)
+        href           = Addressable::URI.parse(attachment['url']).normalize.to_s
+        thumbnail_href = attachment['icon'].is_a?(String) ? attachment['icon'] : attachment.dig('icon', 'url')
```

Comment:
maybe the other way around? `attachment['icon'].is_a?(Hash) ? attachment.dig('icon', 'url') : attachment['icon']`